woocommerce/woocommerce-android
# Issue/5997 Variations bulk update sale price

Note: this PR implements the https://github.com/woocommerce/woocommerce-android/issues/5997 partially and targets the base feature branch. It's a continuation of https://github.com/woocommerce/woocommerce-android/pull/6566 PR and should be merged after that one. Marking it as a draft until then.

### Description
This PR adds support for sale price bulk updating along with a few improvements.
* Allowing to bulk update sale prices.
* Showing progress bar dialog and blocking the UI during the bulk update operation.
* Supporting bulk updating prices to "empty" values

### Testing instructions
1. Open product details and go to "Variations" list (product must have variations defined).
2. After variations are loaded click the "Bulk update..." menu item.
3. The bottom sheet should appear. Choose "Sale price".
4. Enter the new regular price value and click "Done".
5. Price update screen should be closed after successful operation and a confirmation snack should appear. In case of error, the screen should not be closed and the error snack should be shown.

### Visual changes

https://user-images.githubusercontent.com/4527432/170757742-236ae335-05bb-4fed-80db-02f1d35ad4de.mp4
